### ISSUE TITLE
UWP commands: generate binding redirects

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/EntityFramework#5471]
Because UWP commands use netcore50 assemblies but actually execute on net451, we can run into issues with assembly versions.

Example error:

```
PS > Add-Migration Test
System.IO.FileLoadException: Could not load file or assembly 'System.IO.FileSystem.Primitives, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'System.IO.FileSystem.Primitives, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Microsoft.EntityFrameworkCore.Migrations.Design.MigrationsScaffolder.Save(String projectDir, ScaffoldedMigration migration, String outputDir)
   at Microsoft.EntityFrameworkCore.Design.MigrationsOperations.AddMigration(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigration.<>c__DisplayClass0_1.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
```

Other examples: 
- design-time migrations don't fail with a load error, but still fail due to getting the wrong versions of BCL libraries such as DataAnnotations. See https://github.com/aspnet/EntityFramework/issues/5945.
- design-time tools are bound to 1.0.0 assembly versions but need to execute against other versions such as 1.0.1. On UWP, NuGet won't add the necessary binding redirects. See https://github.com/aspnet/EntityFramework/issues/6551.
### Solution

We need to generate binding redirects so UWP tooling can get the right assembly versions
### Workaround

Users can manually generate these. Add a file named "app.config" to the UWP project to explicitly redirect to the right assembly versions.

``` xml
<configuration>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="System.IO.FileSystem.Primitives"
                          publicKeyToken="b03f5f7f11d50a3a"
                          culture="neutral" />
        <bindingRedirect oldVersion="4.0.0.0"
                         newVersion="4.0.1.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Threading.Overlapped"
                          publicKeyToken="b03f5f7f11d50a3a"
                          culture="neutral" />
        <bindingRedirect oldVersion="4.0.0.0"
                         newVersion="4.0.1.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.ComponentModel.Annotations"
                          publicKeyToken="b03f5f7f11d50a3a"
                          culture="neutral" />
        <bindingRedirect oldVersion="4.1.0.0"
                         newVersion="4.0.0.0"/>
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

**Comment by StuartSmith** [comment]
see issue [6551](https://github.com/aspnet/EntityFramework/issues/6551) for solution

**Comment by sjb-sjb** [comment]
The UWP / SQLite getting started page (http://ef.readthedocs.io/en/latest/platforms/uwp/getting-started.html) should be updated to show that redirects for EF core 1.0.0 must be included in the App.config file. In addition, when I tried it the demo required "using EFGetStarted.UWP;" in MainPage.xaml.cs and App.xaml.ca

The necessary redirects (from Librazy's solution) are: 

      <dependentAssembly>
        <assemblyIdentity name="Microsoft.EntityFrameworkCore" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="1.0.0.0" newVersion="1.0.1.0"/>
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.EntityFrameworkCore.Relational" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="1.0.0.0" newVersion="1.0.1.0"/>
      </dependentAssembly>

**Comment by divega** [comment]
Bringing back to 1.1.0-preview1. I made a comment on the PR at https://github.com/aspnet/EntityFramework.Tools/pull/12/files#r83714506 but I fogot github doesn't send it now until you publish a review and apparently I didn't submit it on time for it to be considered. 

I suspect the stopgap fix is missing some important issues we were trying to fix for 1.1.0-preview1.

**Comment by bricelam** [comment]
By hooking `AppDomain.AssemblyResolve`, we're able to remove the need to add binding redirects.
